modules/apps/frontend-js/frontend-js-spa-web/src/main/java/com/liferay/frontend/js/spa/web/internal/FrontendSPAImpl.java

```diff
@@ -0,0 +1,99 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2025 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.frontend.js.spa.web.internal;
+
+import com.liferay.frontend.js.spa.web.internal.configuration.SPAConfiguration;
+import com.liferay.portal.configuration.module.configuration.ConfigurationProvider;
+import com.liferay.portal.kernel.frontend.spa.FrontendSPA;
+import com.liferay.portal.kernel.log.Log;
+import com.liferay.portal.kernel.log.LogFactoryUtil;
+import com.liferay.portal.kernel.module.configuration.ConfigurationException;
+import com.liferay.portal.util.PropsValues;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Bryce Osterhaus
+ */
+@Component(
+	configurationPid = "com.liferay.frontend.js.spa.web.internal.configuration.SPAConfiguration",
+	service = FrontendSPA.class
+)
+public class FrontendSPAImpl implements FrontendSPA {
+
+	@Override
+	public boolean enabled(long siteGroupId) {
+		SPAConfiguration spaConfiguration = _getSPAConfiguration(siteGroupId);
+
+		return spaConfiguration.enabled();
+	}
+
+	private SPAConfiguration _getSPAConfiguration(long siteGroupId) {
+		try {
+			return _configurationProvider.getGroupConfiguration(
+				SPAConfiguration.class, siteGroupId);
+		}
+		catch (ConfigurationException configurationException) {
+			if (_log.isDebugEnabled()) {
+				_log.debug(
```

Comment:
This should be log.error since a configuration exception means something unexpected not a missing configuration (I think).

I mean I would understand a debug level if the exception was due to a missing config but if it is not the case, then it should be error (and maybe state that we will use a default configuration as fallback in the message).